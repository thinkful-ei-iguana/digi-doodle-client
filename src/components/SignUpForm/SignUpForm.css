@font-face {
    font-family: "cf_crayonsregular";
    src: url("../../Fonts/Crayon/cfcrayons-regular-webfont.woff2") format("woff2"),
         url("../../Fonts/Crayon/cfcrayons-regular-webfont.woff") format("woff");
    font-weight: normal;
    font-style: normal;
  }
  body {
    background-image: url("../../Pictures/paper-background.jpg");
    background-position: center;
    background-position-x: 22%;
    background-position-y: 2%;
  }
  .rules-container {
    font-weight: 600;
    background-color: white;
    opacity: 90%;
    border: 2px solid #3ea6fb;
    border-radius: 10px;
  }
  .welcome-header {
    margin-top: 15%;
  }
  .rules-li {
    text-align: left;
    padding-top: 5px;
  }
  .intro-text {
    font-size: 18px;
  }
  .start-button {
    margin-top: 1.5%;
    border-radius: 10px;
    font-size: 30px;
    font-family: "cf_crayonsregular", fallback, sans-serif;
    background-color: #f55440;
    box-shadow: 2px 3px #222222;
    padding-left: 40px;
    margin-right: 1.5%;
    font-weight: 600;
  }
  input {
    border: none;
    border-bottom: 1px solid #222222;
    font-size: 24px;
    width: 175px;
    padding-left: 10px;
    background: none;
    margin-right: 1%;
  }
  .logo {
    margin-top: 2%;
    max-width: auto;
  }
  p {
    margin-top: 5%;
    font-size: 22px;
    margin-right: 1.2%;
    font-weight: 600;
  }
  .error {
    margin-bottom: 2%;
    font-size: 18px;
    padding: 0px 15px;
    color: red;
  }
  .greeting {
    margin-bottom: -3%;
    font-size: 22px;
  }
  @media screen and (max-width: 450px) {
    .logo {
      width: 275px;
      margin-top: 15%;
    }
    .error {
      margin-bottom: -2%;
    }
    .greeting {
      margin-bottom: -10%;
      margin-top: 4%;
    }
    p {
      margin-top: 14%;
    }
    .start-button {
      margin-top: 10%;
    }
    input {
      margin-top: -3%;
    }
    .rules-container {
      margin: 20px 15px;
    }
    .welcome-header {
      margin-top: 5%;
    }
    .rules-li {
      margin-left: -30px;
      margin-bottom: 15px;
      margin-right: 3.5px;
    }
    .intro-text {
      margin-top: 3%;
    }
  }
  @media screen and (min-width: 500px) {
    .rules-li {
      font-size: 18px;
    }
    .rules-container {
      margin: 5% 15px;
    }
    .welcome-header {
      margin-top: 4%;
    }
  }
  @media screen and (min-width: 800px) {
    .welcome-header {
      font-size: 28px;
      margin-top: 5%;
    }
    .intro-text {
      font-size: 24px;
    }
    .rules-li {
      font-size: 18px;
      margin-right: 1%;
    }
    .rules-container {
      margin-bottom: 5%;
    }
    .rules-container {
      margin: 10px 25px;
    }
  }
  @media screen and (min-width: 1000px) {
    .rules-container {
      margin: 20px 15%;
    }
    .welcome-header {
      margin-top: 2%;
    }
    .intro-text {
      margin-top: 1%;
    }
  }
