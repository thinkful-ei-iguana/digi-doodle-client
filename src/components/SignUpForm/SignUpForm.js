import React from 'react';
import Logo from '../../Pictures/digidoodle-logo.png';
import DigiDoodleApiService from '../../services/digi-doodle-api-service';
import ColorContext from '../../Context/ColorContext'
import Cookies from 'js-cookie';
import './SignUpForm.css'

class SignUpForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      greeting: false,
      error: {
        error: ''
      },
      playerId: '',
      gameId: ''
    }
  }

  static contextType = ColorContext;

  handleChange = (event) => {
    this.setState({
      username: event.target.value,
      greeting: true,
    });
  }

  handleSubmit = async (event) => {
    let username = this.state.username;


    if (username.length < 4 || username.length > 10) {
      event.preventDefault();
      return this.setState({
        error: {
          error: 'Please choose a username between 4 and 10 characters'
        }
      })
    }
    event.preventDefault();

    try {
      await this.context.setUserName(this.state.username);
      let userName = this.state.username;

      let userID = await DigiDoodleApiService.createUserName(userName);
      console.log('userId: ', userID);
      userID = userID[0];
      await this.setPlayerId(userID);
      await this.context.setUserId(userID);

      let gameData = await DigiDoodleApiService.joinGame(this.state.playerId, userName);
      console.log('gamedata response: ', gameData);
      gameData = gameData[0];
      await this.setGameId(gameData);
      await this.context.setGameId(gameData);

      let cookieData = {
        username: userName,
        userID: userID,
        gameId: gameData
      }

      Cookies.set('digi-doodle-user', cookieData, { expires: 1 });

      this.props.history.push('/lobby');

    } catch (error) {
      this.setState({
        error: {
          error: error.error
        }
      });
    }
  }

  setPlayerId(uuid) {
    this.setState({
      playerId: uuid
    })
  }

  setGameId(uuid) {
    this.setState({
      gameId: uuid
    })
  }

  render() {
    const username = this.state.username;
    const greeting = this.state.greeting;
    let error = this.state.error.error;
    let message;
    let errorMessage;

    if (error) {
      errorMessage = <h2 className="error">{this.state.error.error}!</h2>
    }
    if (greeting) {
      message = <h2 className="greeting">Hello, {username}!</h2>
    }

    return (
      <div className="sign-up-page">
        <div className="logo-container">
          <img className="logo" src={Logo} alt="logo" />
        </div>
        <div className="sign-up-form-container" onSubmit={this.handleSubmit}>
          <form className="sign-up-form">
            {errorMessage}
            {message}
            <p>Enter your username:</p>
            <input
              type='text'
              onChange={(event) => this.handleChange(event)}
              required
              maxLength="15"
              spellCheck="false"
            />
            <br />
            <button className="start-button" type="submit">Play Game!</button>
          </form>
        </div>
        <h2 className="welcome-header">Welcome to digi-doodle<span className="exclamation">!</span></h2>
        <div className="rules-container">
          <p className="intro-text">The rules of the game are simple:</p>
          <ul className="rules-ul">
            <li className="rules-li">• First enter your desired username to start playing<span className="exclamation">!</span></li>
            <li className="rules-li">• When it’s your turn, you will draw a picture based on the given word or phrase that you see. </li>
            <li className="rules-li">• When you're guessing what another player is drawing, you will type and submit your guess in the chat box below the canvas.</li>
            <li className="rules-li">• The drawer gets 2 points when a player makes a correct guess, and the guesser gets 1 point for each correct guess. The first player to score 15 points wins<span className="exclamation">!</span></li>
          </ul>
        </div>
      </div>
    );
  }
}


export default SignUpForm;