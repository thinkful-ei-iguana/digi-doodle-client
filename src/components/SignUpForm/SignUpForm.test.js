import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import SignUpForm from './SignUpForm'
import { mount } from 'enzyme'


describe(`SignupForm component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <SignUpForm />
            , div);

        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders SignUp form', () => {
        const form = renderer.create(<SignUpForm />).toJSON();
        expect(form).toMatchSnapshot();
    })

    it('renders text input with label', () => {
        const wrapper = mount(<SignUpForm />)
        const input = wrapper.find('input');
        expect(input).toHaveLength(1);
        expect(input.prop('type')).toEqual('text')
    })

})