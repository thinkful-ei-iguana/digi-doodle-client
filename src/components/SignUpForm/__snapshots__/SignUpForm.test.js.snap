// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm component renders SignUp form 1`] = `
<div
  className="sign-up-page"
>
  <div
    className="logo-container"
  >
    <img
      alt="logo"
      className="logo"
      src="digidoodle-logo.png"
    />
  </div>
  <div
    className="sign-up-form-container"
    onSubmit={[Function]}
  >
    <form
      className="sign-up-form"
    >
      <p>
        Enter your username:
      </p>
      <input
        maxLength="15"
        onChange={[Function]}
        required={true}
        spellCheck="false"
        type="text"
      />
      <br />
      <button
        className="start-button"
        type="submit"
      >
        Play Game!
      </button>
    </form>
  </div>
  <h2
    className="welcome-header"
  >
    Welcome to digi-doodle
    <span
      className="exclamation"
    >
      !
    </span>
  </h2>
  <div
    className="rules-container"
  >
    <p
      className="intro-text"
    >
      The rules of the game are simple:
    </p>
    <ul
      className="rules-ul"
    >
      <li
        className="rules-li"
      >
        • First enter your desired username to start playing
        <span
          className="exclamation"
        >
          !
        </span>
      </li>
      <li
        className="rules-li"
      >
        • When it’s your turn, you will draw a picture based on the given word or phrase that you see. 
      </li>
      <li
        className="rules-li"
      >
        • When you're guessing what another player is drawing, you will type and submit your guess in the chat box below the canvas.
      </li>
      <li
        className="rules-li"
      >
        • The drawer gets 2 points when a player makes a correct guess, and the guesser gets 1 point for each correct guess. The first player to score 15 points wins
        <span
          className="exclamation"
        >
          !
        </span>
      </li>
    </ul>
  </div>
</div>
`;
